src/code/z_collision_check.c

```diff
@@ -1,392 +1,3373 @@
+            func_8005B280(globalCtx->state.gfxCtx, &quad->dim.quad[2], &quad->dim.quad[3], &quad->dim.quad[1]);
+            func_8005B280(globalCtx->state.gfxCtx, &quad->dim.quad[1], &quad->dim.quad[0], &quad->dim.quad[2]);
+            break;
+    }
+}
+
+void CollisionCheck_Draw(GlobalContext* globalCtx, CollisionCheckContext* collisionCheckCtx) {
+    Collider* collider;
+    s32 i;
+
+    if (AREG(15)) {
+        if (AREG(21)) {
+            for (i = 0; i < collisionCheckCtx->colAtCount; i++) {
+                func_8005D4DC(globalCtx, collisionCheckCtx->colAt[i]);
+            }
+        }
+        if (AREG(22)) {
+            for (i = 0; i < collisionCheckCtx->colAcCount; i++) {
+                func_8005D4DC(globalCtx, collisionCheckCtx->colAc[i]);
+            }
+        }
+        if (AREG(23)) {
+            for (i = 0; i < collisionCheckCtx->colOcCount; i++) {
+                collider = collisionCheckCtx->colOc[i];
+                if (collider->maskA & 1) {
+                    func_8005D4DC(globalCtx, collider);
+                }
+            }
+        }
+        if (AREG(24)) {
+            func_80042C3C(globalCtx, &globalCtx->colCtx);
+        }
+        if (AREG(25)) {
+            func_80042FC4(globalCtx, &globalCtx->colCtx);
+        }
+    }
+}
+
+s32 (*D_8011DEF8[])(GlobalContext*, Collider*) = { Collider_JntSphSetAT, Collider_CylinderSetAT, Collider_TrisSetAT,
+                                                   Collider_QuadSetAT };
+
+// Sets collider as an AT (attack) for the current frame, allowing it to be checked against ACs (attack colliders)
+s32 CollisionCheck_SetAT(GlobalContext* globalCtx, CollisionCheckContext* collisionCheckCtx, Collider* collider) {
+    s32 index;
+
+    if (func_800C0D28(globalCtx) == 1) {
+        return -1;
+    }
+    if (!(collider->shape < COLTYPE_INVALID)) {
+        __assert("pcl_obj->data_type <= CL_DATA_LBL_SWRD", "../z_collision_check.c", 2997);
+    }
+    D_8011DEF8[collider->shape](globalCtx, collider);
+    if (collider->actor != NULL) {
+        if (collider->actor->update == NULL) {
+            return -1;
+        }
+    }
+    if (collisionCheckCtx->colAtCount >= COLLISION_CHECK_AT_MAX) {
+        osSyncPrintf("CollisionCheck_setAT():インデックスがオーバーして追加不能\n");
+        // EUC-JP: インデックスがオーバーして追加不能 | Index exceeded and cannot be added
+        return -1;
+    }
+    if (collisionCheckCtx->unk2 & 1) {
+        return -1;
+    }
+    index = collisionCheckCtx->colAtCount;
+    collisionCheckCtx->colAt[collisionCheckCtx->colAtCount++] = collider;
+    return index;
+}
+
+// Sets collider as an AT (attack) for the current frame, allowing it to be checked against ACs (attack colliders).
+// If CollisionCheck_SAC is enabled, the collider will be inserted into the list at the specified index, otherwise it
+// will be inserted into the next slot
+s32 CollisionCheck_SetAT_SAC(GlobalContext* globalCtx, CollisionCheckContext* collisionCheckCtx, Collider* collider,
```

Comment:
what does SAC mean in all of these function names? Is there anyway to make it more clear?
Would like to get rid of the extra underscores in all these names if we can.

Reply:
don't really understand what the SAC name stand for specifically. It's an unused system that allows for directly assigning colliders to specific indices of the at/ac/oc lists when CollisionCheckCtx.unk_02 lsb is 1, falling back to normal SetAT/AC/OC behavior except that OC_SAC looks bugged. nothing in the code base uses this, nor should anything use this to set these lists.

Reply:
completley unused, wow ok. A good name is less important then. I think a comment like // unused above all the functions you know dont get used in the game would be a good idea